src/core/Akka.Persistence/Snapshot/LocalSnapshotStore.cs

```diff
@@ -201,10 +201,14 @@ protected virtual void Save(SnapshotMetadata metadata, object snapshot)
             {
                 Serialize(stream, new Serialization.Snapshot(snapshot));
             });
-            tempFile.MoveTo(GetSnapshotFileForWrite(metadata).FullName);
+		    var newName = GetSnapshotFileForWrite(metadata);
```

Comment:
I dont classify this as a hack. Its simply the result of a difference in SDK implementation between JVM and .net imho